ledger/ledger-utxo/src/main/java/net/corda/v5/ledger/utxo/ContractVerificationException.java

```diff
@@ -0,0 +1,84 @@
+package net.corda.v5.ledger.utxo;
+
+import net.corda.v5.base.exceptions.CordaRuntimeException;
+import net.corda.v5.crypto.SecureHash;
+import org.jetbrains.annotations.NotNull;
+
+import java.util.List;
+
+/**
+ * Represents the exception that is thrown for a contract verification failure.
+ */
+public final class ContractVerificationException extends CordaRuntimeException {
+
+    /**
+     * The transaction ID of the transaction that has failed verification.
+     */
+    @NotNull
+    private final SecureHash transactionId;
+
+    /**
+     * The list of failure reasons for the transaction that has failed verification.
+     */
+    @NotNull
+    private final List<ContractVerificationFailure> failureReasons;
+
+    /**
+     * Creates a new instance of the {@link ContractVerificationException} class.
+     *
+     * @param transactionId  The transaction ID of the transaction that has failed verification.
+     * @param failureReasons The list of failure reasons for the transaction that has failed verification.
+     */
+    public ContractVerificationException(
+            @NotNull final SecureHash transactionId,
+            @NotNull final List<ContractVerificationFailure> failureReasons) {
```

Comment:
@lankydan I think I'm getting what you mean, as per the previous comment too...

Do we have a style preference for this? Personally I think a new line containing `) {` looks terrible, mostly because the Java stylistic convention for `{` is at the end of a line, not on a new line (which in this case could be argued both ways).

```java
public ContractVerificationException(
    @NotNull final SecureHash transactionId,
    @NotNull final List<ContractVerificationFailure> failureReasons
) {
    this.transactionId = transactionId;
    this.failureReasons = failureReasons;
}
```

vs.

```java
public ContractVerificationException(
        @NotNull final SecureHash transactionId,
        @NotNull final List<ContractVerificationFailure> failureReasons) {
    this.transactionId = transactionId;
    this.failureReasons = failureReasons;
}
```

I personally think the latter looks cleaner, but if there's a consensual preference towards the former, I'l change it.